docs/reference/modules/piece-type.md

```diff
@@ -0,0 +1,272 @@
+---
+extends: '@apostrophecms/doc-type'
+---
+
+# `@apostrophecms/piece-type`
+
+This module is the foundation for all [piece types](/guide/pieces.md) in Apostrophe. It is not typically configured or referenced in project code directly since each piece type should be managed independently in most cases. For example, the options documented below would be configured on a custom piece type, e.g., `article`, rather this piece type base module.
+
+The only reason to configure this module directly would be to apply the changes to *every* piece type, including those Apostrophe core (e.g., `@apostrophecms/user`).
+
+**Extends:** `{{ $frontmatter.extends }}`
+
+## Options
+
+|  Property | Type | Description |
+|---|---|---|
+| [`autopublish`](#autopublish) | Boolean | Set to `true` to publish all saved edits immediately. |
```

Comment:
Hmm in the text below I think you did cover the defaults.

Reply:
A lot of them don't have true defaults and are instead left to `undefined`. It could be good self-documentation to have `autopublish: false` in the piece type module.

Reply:
That's true.

Reply:
A lot of them get their default from an `||` at the time they are used. Explicit defaults are better.